### ISSUE TITLE
Allow autoscaling based on local static provisioned volumes

### ISSUE THREAD

**Issue by kanor1306** [kubernetes-sigs/sig-storage-local-static-provisioner#288]
**Is your feature request related to a problem? Please describe.**
To summarise: in an autoscaling scenario the autoscaler cannot scale based in volumes provisioned by the local static provisioner, as the usage of `provisioner: "kubernetes.io/no-provisioner"` prevents the autoscaling from been triggered due to how the [scheduler binder](https://github.com/kubernetes/kubernetes/blob/473af0b8d1951b5c362df60925bb8d9fb007fab9/pkg/controller/volume/scheduling/scheduler_binder.go#L868) works

The full problem is described in [the autoscaler repository](https://github.com/kubernetes/autoscaler/issues/1658#issuecomment-1036205889) by myself. But after thinking about it, I think it somehow makes more sense that it is solved in the provisioner side.

**Describe the solution you'd like in detail**
First of all, I have no insight on how CSI drivers work, but a solution to this issue could be that, optionally, the local provisioner could deploy a simple csidriver, that somehow allowed the PVCs to be bound to the PV created by the static provisioner. My guess is that the driver should "inform" the PVC about the name of the newly created PV, but this is just a guess.

**Describe alternatives you've considered**
Introducing another keyword instead of `provisioner: "kubernetes.io/no-provisioner"` that could be used to indicate to the scheduler binder that the volume will exist, even although it does not exist now (like it does with dynamic provisioners). This would need to be changed in the Kubernetes scheduler binder first, and then introduced in the local static provisioner.

**Additional context**
Full context in the mentioned comment and some previous comments in that thread: https://github.com/kubernetes/autoscaler/issues/1658#issuecomment-1036205889

**Comment by Kaitou786** [comment]
@kanor1306 we are facing this same problem; are you currently using any kind of workaround for this problem?
Other than creating a different pod to force autoscale?

**Comment by kanor1306** [comment]
> @kanor1306 we are facing this same problem; are you currently using any kind of workaround for this problem? Other than creating a different pod to force autoscale?

@Kaitou786 actually I stopped using this provisioner for local disks and moved to https://openebs.github.io/lvm-localpv. 

I have to say though, that has been a while since I implemented and used these local volumes with the openebs lvm-localpv, but I remember it also had its own problems. I remember that I did manage to make it scale up and down, but not very smoothly, and some times it would get stuck in one single extra node. But this was one year ago, and on older versions of Kubernetes, so maybe now it works better (but as said, we finally opted out from using local volumes due to other reasons)
